hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/predicate/CaseExpressionTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.predicate;
+
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.ConstantExpression;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionTestBase;
+import com.hazelcast.sql.impl.expression.SimpleExpressionEvalContext;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.sql.impl.type.QueryDataType.BIGINT;
+import static com.hazelcast.sql.impl.type.QueryDataType.BOOLEAN;
+import static com.hazelcast.sql.impl.type.QueryDataType.INT;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class CaseExpressionTest extends ExpressionTestBase {
```

Comment:
We can add this test:
```java
    @Test
    public void testMixedTypes() {
        SqlService sql = createEndToEndRecords();
        assertQueryThrows(sql, "select __key from records where case boolean1 when true then int1 else boolean1 end",
                "Illegal mixing of types in CASE or COALESCE statement");
    }
```

Reply:
case removed in c1a5855c8212e429b5d7245a1906713af01e624b and 7f29e69e24bcd44dec551b9d255171a48bdda0ec, https://github.com/hazelcast/hazelcast/issues/17329 created